### DISCUSSION TITLE
[Feature Request] Ability to provide custom/wrapped TableRow components to a table (and allow other child-bearing list components to be given custom wrappers)

### DISCUSSION THREAD

**Discussion by MarcusDelvecchio** [heroui-inc/heroui#4662] [Feature requests]
### Is your feature request related to a problem? Please describe.

I have been looking into and following related issues involving the limitations on wrapping TableRow components or other child-rendered components because of [tradeoffs within the library's internal implementation details](https://github.com/nextui-org/nextui/issues/729#:~:text=just%20a%20technical%20trade%20off%20required%20by%20internal%20implementation%20details ).

However, with recent [updates to the React Aria library](https://github.com/nextui-org/nextui/issues/729#issuecomment-2269371223) by @devongovett it seems like there might finally be the possibility of this feature/logic being developed.

### Describe the solution you'd like

As a user, I would like to be able to wrap the TableRow component in a custom component and provide it as a child to the TableBody, as a means of providing custom logic (such as expand-on-click or other state-dependant logic) to the rows within a table.

### Describe alternatives you've considered

I have considered adding a 'Actions' column with a button that could trigger an appearing column, I have considered using a popover to display the content I wish to display in my expanding row, but would really love to see this core functionality built in.

### Screenshots or Videos

<img width="497" alt="image" src="https://github.com/user-attachments/assets/57559ffc-eaa4-400a-b535-f5b9c788fee2">

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1273/[feature-request]-expandable-table-rows-and-providing-custom-table-row">ENG-1273 [Feature Request] Expandable table rows and providing custom table row (and other child) components</a></p>

**Comment by miniluz** [comment]
Any updates or timeline on this? This is completely stopping me from using NextUI's table component

**Comment by mbUSC** [comment]
Any updates? It's also blocking me from using NextUI Listbox

**Reply by dhananjaykumar880** [reply]
Any update? it is worst to think dynamic logic using heroui Listbox.
